hw_diag/tasks.py

```diff
@@ -39,7 +40,20 @@ def perform_hw_diagnostics(ship=False):  # noqa: C901
     diagnostics['AN'] = public_keys['name']
 
     # Fetch data from miner container.
-    diagnostics = fetch_miner_data(diagnostics)
+    try:
+        diagnostics = fetch_miner_data(diagnostics)
+    except MinerFailedFetchData as e:
+        if 'MC' not in diagnostics or ('MC' in diagnostics and diagnostics['MC'] <= 0 or diagnostics['MC'] is None):
```

Comment:
There are two things that can be improved about this pattern:

1. The `if` statements should be removed. Even if previous values for these exist, we should record the failure when it happens.
2. We need to set values for more keys than just `MC` and `MN`. There are [other variables ](https://github.com/NebraLtd/hm-diag/pull/212/files#diff-57790e52a73cdf1e8700d99bf6714a8bb8cfbd3268f66bc652d8a74bba9b581aL15)that are normally set too.

```
    diagnostics['MC'] = peerbook.get('connection_count') > 1
    diagnostics['MD'] = peerbook.get('listen_addr_count') > 0
    diagnostics['MH'] = height.get('height')
    diagnostics['MN'] = peerbook.get('nat')
    diagnostics['MR'] = diagnostics['MN'] == 'symmetric'
```

Reply:
Perfect, I'll set all the values to the error message. Would it make more sense to set the values to the simple message I currently have or set it to the exception itself

Reply:
With the code as-is, there's actually a downstream implication for these values. For example, MH is later used to determine the value for MS. So setting MH to -1 would make more sense than as an error message. After looking at the code, I think MH is the only field where we need to be careful.

Reply:
I pushed the changes for this @marvinmarnold 

Would it make sense to set MC, MD, and MR to false as they seem to be conditionals or should a string with the error message be alright here, from what I've seen it should be fine as string here since MR will be set to false automatically since MN is the error string.

Reply:
Good thought, I'm not sure. @ilyastrodubtsev will these values have any impact on what gets displayed on the dashboard?